### ISSUE TITLE
Do not Steal Focus on Job Completion

### ISSUE THREAD

**Issue by Sledge-Hammer** [staxrip/staxrip#333]
Hello,

When a task completes (demuxing, encoding, etc) Stax steals the focus from the currently active window.

This is a huge problem for me because I'm using the PC for other tasks while Stax is working (usually typing something) and 1) it's disruptive and distracting (I want to take a look at it when I'm able to, not when it says so) and 2) if it happens while I'm typing I might be pressing Enter at the same time and close whatever extra dialogue box is displayed (it just happened a few minutes ago and I had to restart the whole process).

I've searched through all the options but couldn't find anything related to this, so perhaps it could be added as a setting for the main window to stay minimized on task completion?

Thanks in advance!

**Comment by stax76** [comment]
There is already a feature that tries exactly to prevent this and it's enabled by default, I just tried it and it's not working any longer, needs investigation.

**Comment by Arathen** [comment]
I may be doing something wrong, but this is happening to me now (v2.6.0). I've looked for the setting/feature that you mentioned above to make sure its enabled, but cant see anything that looks right. Has this bug been reintroduced? Or am I misconfigured somewhere?

**Comment by Dendraspis** [comment]
@Arathen 
No, you're right ... partly. The implemented feature has a time limiter set at 60s fix. After these 60s the Main window opens again on top with stealing the focus from other apps.

I think I might change it a bit for the next release...

**Comment by Dendraspis** [comment]
Done...  😁

**Comment by crisc2000** [comment]
> There is already a feature that tries exactly to prevent this and it's enabled by default, I just tried it and it's not working any longer, needs investigation.

So what are the settings exactly for the StaxRip to stop popup over other windows?
This behavior is very annoying. I use v2.10 with the settings of v2.7 and it's stealing focus by default. here is a print screen with my default options.

![StaxRip focus options](https://user-images.githubusercontent.com/1497398/170862460-b2f0b248-f42a-40af-b6e5-11ea27ff344a.jpg)

Should I set both options to -1 ?

**Comment by AndrewKeeley** [comment]
This still seems to to not work in v2.20.0.

**Comment by Dendraspis** [comment]
I'll take a look at it when I have a bit more time. Haven't experienced any issues last time I used it while doing other stuff.

**Comment by crisc2000** [comment]
Maybe it has something to do with OS version? I use Windows 7 x64.

**Comment by Dendraspis** [comment]
Possible. Or some other settings (outside of StaxRip). But I also guess it could be related to the app in foreground.
I'll try to reproduce it later....

**Comment by AndrewKeeley** [comment]
Thank you. I'm using W10 Pro x64 22H2.

It does not appear to be related to the foreground apps. I see it with Visual Studio, Excel, Word, Notepad, Media Player and others. I do not recall ever using a program that wasn't affected. It would be ideal if StaxRip could just stay minimized until the user chooses to restore it. Perhaps flash the taskbar icon when the queue is finished.

**Comment by Dendraspis** [comment]
I'm using Win10 Pro 20H2 and I'll test it tonight or tomorrow.
Did you guy use the default settings for the prevention or did you modify the values?

**Comment by AndrewKeeley** [comment]
> Did you guy use the default settings for the prevention or did you modify the values?

Both. I tried all sorts of values.

**Comment by Dendraspis** [comment]
Okay, I can reproduce the (wrong) behavior. But it seems to be not that easy to fix.  😕

**Comment by Dendraspis** [comment]
What should I say .... it is fixed.  🥳 
Next release is coming Monday...

**Comment by AndrewKeeley** [comment]
> What should I say .... it is fixed. 🥳 Next release is coming Monday...

Awesome! Thank you.

**Comment by AndrewKeeley** [comment]
v2.21 seems to have fixed the problem, as long as the window is minimized. If it's left open it still takes focus when switching stages, but I can easily live with that! This is a huge improvement for me. Thanks again!

**Comment by Dendraspis** [comment]
Hmm, I've never minimized it while testing, just let it open,  brought other app to front and waited.  🤔

Could be related to the apps or OS. I might finetune it, if possible, when I update my OS someday, but at the moment the is nothing I can do.

Btw: you use a clean setup without copying an older settings file, right?

**Comment by AndrewKeeley** [comment]
I copied the entire settings folder from v2.20 because I have so many profiles created, but I will try it tomorrow with a fresh settings.dat file.

**Comment by Dendraspis** [comment]
Ok, that could explain it. I'm curious to read your feedback.

**Comment by Dendraspis** [comment]
🤔  I guess you have an audio conversion going on? Can you select `No Audio` please and check, if that makes a difference for you?
Is the processing window opening shortly for a few seconds and then the main window is opened in background or does the main window open and takes the focus?

**Comment by Dendraspis** [comment]
I think I have found what you have meant. We will see what you experience with the next release....

**Comment by AndrewKeeley** [comment]
Sounds good. Like I said, I am happy to minimize it when I'm working on other things, and then it's fine anyway.

**Comment by Dendraspis** [comment]
Yeah, but now I'm in ***Challenge accepted*** mode.  😂 
Please test also the new release, should also work without minimizing.  🤓

**Comment by AndrewKeeley** [comment]
I tried v2.22 with a fresh settings.dat. I ran one job in batch mode leaving StaxRip open. As soon as I clicked start, I switched to another application. When ffmpeg opened to demux, it took focus. When ffmpeg and nvencc64 opened at the same time to start encoding, they took focus. When nvencc64 finished and the tab closed leaving just ffmpeg running, it took focus. When qaac started, it took focus. When mp4box started, it took focus. When mp4box finished and the main window reappeared, it did NOT take focus.
